lib/redis_memo/memoize_method.rb

```diff
@@ -81,14 +81,50 @@ def self.extract_dependencies(ref, *method_args, &depends_on)
 
   def self.get_or_extract_dependencies(ref, *method_args, &depends_on)
     if RedisMemo::Cache.local_dependency_cache
-      RedisMemo::Cache.local_dependency_cache[ref] ||= {}
-      RedisMemo::Cache.local_dependency_cache[ref][depends_on] ||= {}
-      RedisMemo::Cache.local_dependency_cache[ref][depends_on][method_args] ||= extract_dependencies(ref, *method_args, &depends_on)
+      RedisMemo::Cache.local_dependency_cache[ref.class] ||= {}
+      RedisMemo::Cache.local_dependency_cache[ref.class][depends_on] ||= {}
+      mapped_args = map_depends_on_method_args(depends_on, ref, method_args)
+      RedisMemo::Cache.local_dependency_cache[ref.class][depends_on][mapped_args] ||= extract_dependencies(ref, *method_args, &depends_on)
     else
       extract_dependencies(ref, *method_args, &depends_on)
     end
   end
 
+  # We only look at method parameters that the dependency block cares about in order to define its dependent
+  # memos, following the convention that nil or :_ means we don't care about the parameter.
+  # Example:
+  # ```
+  #    def method(param1, param2)
+  #    end
+  #
+  #    memoize_method :method do |_, _, param2|`
+  #      depends_on RedisMemo::Memoizable.new(param2: param2)
+  #    end
+  # ```
+  #  `map_depends_on_method_args(depends_on, ref, [1, 2])` returns { param2 : 2 }
+  def self.map_depends_on_method_args(depends_on, ref, args)
+    mapped_args = {}
+    unless depends_on.parameters.empty? or args.empty?
+      depends_on_args = [ref] + args
+      depends_on_parameters = depends_on.parameters.clone
+      # A double splat should always be the last parameter
+      if depends_on_parameters[-1][0] == :keyrest && depends_on_args[-1].is_a?(Hash)
+        mapped_args[depends_on_parameters.pop[1]] = depends_on_args.pop
```

Comment:
`pop[1]` can this be nil?

Reply:
`depends_on_parameters.pop` will get something like `[:keyrest, :kwargs]`, the [1] index I think should just get the parameter name.

Reply:
but one can still have something like this right?
```
irb(main):001:0> def a(*, **); end
=> :a
irb(main):002:0> method(:a).parameters
=> [[:rest], [:keyrest]]
```

Reply:
Ah yea you're right, I should have checked more thoroughly but this is also why the current specs are failing. Will fix